src/com/elixr/training/SearchWord.java

```diff
@@ -0,0 +1,42 @@
+package com.elixr.training;
+import static com.elixr.training.Main.*;
+import java.sql.SQLException;
+import java.util.StringTokenizer;
+
+ public class SearchWord extends Thread {
+    String data;
+    int count;
+
+    public SearchWord(String data, int count) {
+        this.data = data;
+        this.count = count;
+    }
+
+    public void run() {
+        StringTokenizer st = new StringTokenizer(data);
+
+        while (st.hasMoreTokens()) {
+            if (searchWord.equalsIgnoreCase(st.nextToken())) {
+                count++;
+            }
+        }
+        if (count == 0) {
+            System.out.println("Word not found");
+            try {
+                database.dbOperation(inputFilePath, searchWord, Constants.status, count);
```

Comment:
Accessing the database object from the Main class is not a proper way to do it, you should receive it as an input parameter in the constructor and have a field for it in the SearchWord class itself.

Reply:
This is called **Dependency Injection**

Reply:
I have called it in Searchword class using constructor